### DISCUSSION TITLE
mamba install error

### DISCUSSION THREAD

**Discussion by rtaylo45** [idaholab/moose#20187] [Q&A Getting Started] [answered]
Hello,

I am reinstalling conda/mamba from scratch on a remote machine. I have installed conda and i installed mamba using conda with the following commands `conda install mamba -n base -c conda-forge`. I have added the inl channel and when i try to install `moose-tools` and `moose-libmesh` but i get the following error. 
```
Looking for: ['moose-tools', 'moose-libmesh']

pkgs/main/linux-64       [====================] (00m:00s) No change
pkgs/main/noarch         [====================] (00m:00s) No change
pkgs/r/linux-64          [====================] (00m:00s) No change
pkgs/r/noarch            [====================] (00m:00s) No change
public/linux-64          [====================] (00m:00s) No change
public/noarch            [====================] (00m:00s) No change
Encountered problems while solving:
  - nothing provides gxx_linux-64 9.3.0 h0d07fa4_29 needed by moose-tools-2021.07.13-py39haff3c87_0
  - nothing provides libpng 1.6.37 hed695b0_2 needed by moose-libmesh-2021.06.30-build_1
 ```

Running `mamba list` has the following packages. 
```
# packages in environment at /home/qt5/miniconda3/envs/moose:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main
_openmp_mutex             4.5                       1_gnu
_sysroot_linux-64_curr_repodata_hack 3                   haa98f57_10
binutils_impl_linux-64    2.35.1               h27ae35d_9
binutils_linux-64         2.35.1              h454624a_30
gcc_impl_linux-64         9.3.0               h6df7d76_17
gcc_linux-64              9.3.0               h1ee779e_30
gxx_impl_linux-64         9.3.0               hbdd7822_17
gxx_linux-64              9.3.0               h7e70986_30
kernel-headers_linux-64   3.10.0              h57e8cba_10
ld_impl_linux-64          2.35.1               h7274673_9
libgcc-devel_linux-64     9.3.0               hb95220a_17
libgcc-ng                 9.3.0               h5101ec6_17
libgomp                   9.3.0               h5101ec6_17
libstdcxx-devel_linux-64  9.3.0               hf0c5c8d_17
libstdcxx-ng              9.3.0               hd4cf53a_17
sysroot_linux-64          2.17                h57e8cba_10
```

Thanks,
Zack

**Comment by milljm** [comment, marked as answer]
Interesting. These are very old versions. We're certainly into the early 2022's on moose-libmesh. What instructional page were you following? Perhaps we have a bad page lingering around that we need to deal with!

**Reply by GiudGiud** [reply]
Installation from scratch shouldnt lead to an old version, this is indeed interesting.
@milljm what's the latest version of moose-libmesh? Having an update issue as well and I want to try a new solution.

**Reply by rtaylo45** [reply]
@milljm I went to the moose install doc and saw that some changes were made. Mainly that there is a direct install of mamba and not an install of conda first then having conda install mamba. I thought this was kinda weird so i went to the mamba site. They said that i could still install mamba using conda so i just installed conda then used conda to install mamba. I could have messed up by installing an old version of conda? Im gonna try installing conda again to fix this issue. I did see on the moose install guide that they still say conda when referring to a lot of things, which i think is a typo considering they moose now recommends a direct install of mamba. 

Is there a reason that moose now recommends directly installing mamba and not conda?

**Reply by rtaylo45** [reply]
Ok so i followed the install guide to the T and it looks like everything installed right. I still think this is kinda a strange error but i guess this is why you follow the install guide.

**Reply by milljm** [reply]
Oh I see, this is truly a 're-install' attempt. I think I got confused with the phrase 'from scratch' :)

In this case, I am willing to bet there is a pre-existing 'moose' Conda environment stuck on some old versions of our packages. It happens, and not your fault! What I'd like you to do, is delete this stuck environment:
```bash
conda activate base
conda env remove -n moose
```
and then create it again.
```bash
mamba create -n moose moose-libmesh moose-tools
```
I believe that will bring you up to latest versions.
